src/server/api/controllers/products.controller.js

```diff
@@ -78,19 +57,48 @@ const getProductById = async (id) => {
         'is_on_discount',
         'discount_percent',
         'picture',
-      )
-      .where({ id });
-    if (products.length === 0) {
-      throw new Error(`incorrect entry with the id of ${id}`, 404);
+    )
+    .where('products.is_on_discount', '=', '1');
+
+const getProductById = async(prod_id) => {
```

Comment:
Nitpick(naming):
Avoid short names, `product_id` is not long and using tools like VSCode will give you typing support so there is no reason to offer less information and clarity in your variable names.